server/tests/helpers/user_helper.js

```diff
@@ -0,0 +1,21 @@
+import { fixtures, build } from '../factories';
+
+function buildUserAndGroups() {
+  return new Promise((resolve, reject) => {
+    const generatedValues = {};
+    const b1 = build('group', fixtures.group());
+    const b2 = build('group', fixtures.group());
+    Promise.all([b1, b2])
+      .then((values) => {
+        generatedValues.groups = values;
+        const groupNames = values.map(g => g.name);
+        build('user', fixtures.user({ groups: groupNames }))
```

Comment:
You can return this and then continue to chain .then() on the upper level Promise.all(). You can also return the Promise.all() and it will return a promise. So you won't have to create another one.